src/features/chain-connection/WalletConnection.tsx

```diff
@@ -1,27 +1,26 @@
-import { useAccount, useConnect, useEnsName } from "wagmi";
+import { useAccount, useEnsName, useDisconnect } from "wagmi";
 import { Button } from "../common/Button";
-import { InjectedConnector } from "wagmi/connectors/injected";
 import { twJoin } from "tailwind-merge";
 import truncateEthAddress from "truncate-eth-address";
+import { FC, Dispatch } from "react";
 
-// TODO: this component should actually open a modal instead of defaulting to
-// the injected connector
+interface WalletConnectionProps {
+  setRenderWalletConnectModal: Dispatch<boolean>
+}
 
-export const WalletConnection = ({}: {}) => {
+export const WalletConnection: FC<WalletConnectionProps> = ({ setRenderWalletConnectModal }) => {
```

Comment:
Just a preference, but personally I think it's fine for the `WalletConnection` component to include the modal.

Here you have separated it out into two components, but you are always requiring the function to show the wallet connect modal, implying they'll never be used separately.

Reply:
@personnamedmike on this I didn't actually mean to remove the `WalletConnectionModal` entirely, moreso have the `<WalletConnectionModal />` within the `<WalletConnection />` component.

I'm not used to using the `dialog` element, though, I guess you would need to use `forwardRef` (forwarding to the modal) so you can open the modal from outside that component?

@piersss may have an idea here as he's used this element before.

Reply:
@gpxl-dev passing a `useState` variable and setter works too

Reply:
I see what you mean about useRef